Source/Inspector/InspectorWindow.cpp

```diff
@@ -76,22 +78,19 @@ void InspectorWindow::ShowMeshRenderComponent(Scene& scene)
 		ImGui::SameLine();
 		ImGui::InputText("##Mesh Name",meshBuffer,64);
 		ImGui::SameLine();
+
 		if(ImGui::Button("Choose mesh"))
 		{
-			// TODO: When file manager added, use file manager to select a mesh
-			ImGuiFileDialog::Instance()->OpenDialog("getFileName","Choose Mesh",".obj",".");
+			ImGui::OpenPopup("Select Mesh");
 		}
-	}
 
-	if(ImGuiFileDialog::Instance()->Display("getFileName", ImGuiWindowFlags_NoCollapse, FileDialog::MIN_SIZE, FileDialog::MAX_SIZE))
-	{
-		if(ImGuiFileDialog::Instance()->IsOk())
+		if(bool isOpen = _assetHelper->ShowGuiWindow(); _assetHelper->GetMesh() != nullptr&&isOpen == true)
```

Comment:
_assetHelper->GetMesh() != nullptr is probably not a good idea.
You can't set the mesh to "none".